src/DotNetWorker.Grpc/Handlers/InvocationHandler.cs

```diff
@@ -107,14 +107,21 @@ public async Task<InvocationResponse> InvokeAsync(InvocationRequest request)
                 response.Result.Status = StatusResult.Types.Status.Success;
             }
             catch (Exception ex)
-            {
-                response.Result.Exception = ex.ToRpcException();
-                response.Result.Status = StatusResult.Types.Status.Failure;
+            {   
+                // If this isn't a task canceled exception, it was thrown by user's code. 
 
                 if (ex.InnerException is TaskCanceledException or OperationCanceledException)
                 {
                     response.Result.Status = StatusResult.Types.Status.Cancelled;
                 }
+                else
+                {
+                    response.Result = new StatusResult
```

Comment:
Do we need to add this `else` statement here? Can we just update line 111 to be 

`response.Result.Exception = ex.ToUserRpcException()`

Reply:
Ah wait, I see you dont want it to be a customer error if cancellation happens

Reply:
I think in the case that it's a TaskCanceledException, we'd want to avoid flagging it as thrown by the user's code.

Reply:
In this case we will want to make sure we set `response.Result.Exception = ex.ToRpcException();` for cancellation